Spigot-Server-Patches/0235-Anti-Xray.patch

```diff
@@ -0,0 +1,1477 @@
++                        next = nextNext;
++                        nextNext = temp;
++
++                        for (int z = 0; z < 16; z++) {
++                            for (int x = 0; x < 16; x++) {
++                                if (!solidGlobal[Block.REGISTRY_ID.getId(aboveChunkSection.getType(x, 0, z))]) {
++                                    current[z][x] = true;
++                                }
++                            }
++                        }
++
++                        //There is nothing to read anymore
++                        dataBitsReader.setBitsPerValue(0);
++                        solid[0] = true;
++                        counter = obfuscateLayer(15, dataBitsReader, dataBitsWriter, solid, obfuscateTemp, predefinedBlockDataBits, current, next, nextNext, nearbyChunkSections, counter);
++                    }
++                } else {
++                    //If so, initialize the reader and other stuff for the chunk section above to obfuscate the upper layer of the current chunk section
++                    dataBitsReader.setBitsPerValue(packetPlayOutMapChunkInfoAntiXray.getBitsPerValue(chunkSectionIndex + 1));
++                    dataBitsReader.setIndex(packetPlayOutMapChunkInfoAntiXray.getDataBitsIndex(chunkSectionIndex + 1));
++                    solidTemp = readDataPalette(packetPlayOutMapChunkInfoAntiXray.getDataPalette(chunkSectionIndex + 1), solid, solidGlobal);
++                    obfuscateTemp = readDataPalette(packetPlayOutMapChunkInfoAntiXray.getDataPalette(chunkSectionIndex + 1), obfuscate, obfuscateGlobal);
++                    boolean[][] temp = current;
++                    current = next;
++                    next = nextNext;
++                    nextNext = temp;
++                    counter = obfuscateLayer(15, dataBitsReader, dataBitsWriter, solidTemp, obfuscateTemp, predefinedBlockDataBits, current, next, nextNext, nearbyChunkSections, counter);
++                }
++
++                dataBitsWriter.finish();
++            }
++        }
++
++        packetPlayOutMapChunkInfoAntiXray.getPacketPlayOutMapChunk().setReady(true);
++    }
++
++    private int obfuscateLayer(int y, DataBitsReader dataBitsReader, DataBitsWriter dataBitsWriter, boolean[] solid, boolean[] obfuscate, int[] predefinedBlockDataBits, boolean[][] current, boolean[][] next, boolean[][] nextNext, ChunkSection[] nearbyChunkSections, int counter) {
++        //First block of first line
++        int dataBits = dataBitsReader.read();
++
++        if (nextNext[0][0] = !solid[dataBits]) {
++            dataBitsWriter.skip();
++            next[0][1] = true;
++            next[1][0] = true;
++        } else {
++            if (nearbyChunkSections[2] == Chunk.a || !solidGlobal[Block.REGISTRY_ID.getId(nearbyChunkSections[2].getType(0, y, 15))] || nearbyChunkSections[0] == Chunk.a || !solidGlobal[Block.REGISTRY_ID.getId(nearbyChunkSections[0].getType(15, y, 0))] || current[0][0]) {
```

Comment:
As mentioned above, can you replace all the Chunk.a in this method with a Chunk.EMPTY